File: Crazy8s/Crazy8s/Crazy8ProvidedCode/JavaCrazy8/HumanPlayer (2016_10_11 04_07_57 UTC).java
import java.util.Scanner;
/**
 * Interact with the human player to obtain their desired play.
 */
public class HumanPlayer extends Player {
  
  public HumanPlayer(Deck deck) {
    super(deck);
  }
  
  /**
   * Allow the human player to take a turn by either
   * selecting a card to play to the discard pile from their hand or
   * drawing a card from the deck.
   */
  public void takeATurn(Deck deck, Pile pile) {
    System.out.println("Your hand: ");
    for (int i=0; i<list.size(); i++) {
      System.out.println(i+1 + ": " + list.get(i));
    }
    System.out.println("Top card of pile: " + pile.getTopCard());
    int number;
    Scanner keyboard = new Scanner(System.in);
    boolean firstTime = true;
    do {
      if (firstTime) {
        firstTime = false;
      }
      else {
        System.out.println("Bad input. Please try again.");
      }
      System.out.print("Enter 0 to draw, otherwise number of card to play: ");
      number = keyboard.nextInt();
    } while (number < 0
               ||
             number > list.size()
               ||
             !(number == 0 || pile.isValidToPlay(list.get(number-1))));
    if (number == 0) {
      list.add(deck.dealACard());
    }
    else {
      Card card = list.get(number-1);
      remove(list.indexOf(card));
      pile.acceptACard(card);
      if (card.getValue().equals("8")) {
        String suit;
        do {
          System.out.println("What suit would you like the 8 to represent");
          System.out.print(" (club, diamond, heart, or spade)?");
          suit = keyboard.next();
        } while (!(suit.equals("club") || suit.equals("diamond") ||
                   suit.equals("heart") || suit.equals("spade")));
        pile.setAnnouncedSuit(suit);
      }
    }
  }
}

File: Crazy8s/Crazy8s/HumanPlayer (2016_10_11 04_07_57 UTC).js
"use strict";

function HumanPlayer(deck){
    Player.call(this,deck);
}

HumanPlayer.prototype = Object.create(Player.prototype);

HumanPlayer.prototype.srchHand = function(arr, elt) {
    var ind = -1;
    for(var i = 0; i < arr.length; i++){
	if(elt == arr[i]){
	    ind = i;
	    break;
	}
    }
    
    return ind;
};

HumanPlayer.prototype.takeATurn = function(deck, pile){
    var firstTime = true;
    do{
	if(firstTime == false)
	    window.alert("Bad input. Please try again.");
	var number = window.prompt("Your hand: " + this.list.toString() + "\nTop card of pile: " + pile.getTopCard() + "\nEnter 'p' to draw, otherwise card to play: ");
	if(firstTime)
	    firstTime = false;
	{
	    var card;
	    if(number.length == 2){
		card = new Card(number.slice(1,2), number.slice(0,1));
	    }
	    else if(number.length == 3){
		card = new Card(number.slice(2,3), number.slice(0,2));
	    }
	    else if(number == "p"){
		break;
	    }
	}
    }
    while(HumanPlayer.prototype.srchHand(this.list,number) == -1 ||  !(number == "p"  || (pile.isValidToPlay(card))));

    if(number == "p")
	this.list.unshift(deck.dealACard());
    else{
//	var card = this.list[this.list.indexOf(number)];
	this.remove(HumanPlayer.prototype.srchHand(this.list,number));
	pile.acceptACard(card);
	if(card.getValue() === "8"){
	    var suit;
	    do{
		suit = prompt("What suit would you like the 8 to represent: " + "\nc, d, h, or s?")
	    }
	    while(!(suit === "c" || suit === "d" || suit === "h" || suit === "s"));
	    pile.setAnnouncedSuit(suit);
	}
    }
}
